Makefile

```diff
@@ -99,3 +110,143 @@ verify-shellscripts-initramfs:
 		--severity error \
 		defaults/linuxrc \
 		defaults/initrd.scripts
+
+out/%/:
+	install -d $@
+
+out/temp/%: out/temp/
+	echo > $@
+
+out/build-config: out/temp/genkernel_conf out/temp/man_genkernel_8 out/temp/parse_cmdline \
+out/temp/longusage out/temp/append_base_layout out/temp/create_initramfs \
+out/temp/initramfs_append_func out/temp/determine_real_args
+ifdef GK_FEATURES
+	awk -f compile_features.awk $(foreach feature,${GK_FEATURES},features/$(feature))
+endif
+	echo ${PREFIX} > out/PREFIX
+	echo ${BINDIR} > out/BINDIR
+	echo ${SYSCONFDIR} > out/SYSCONFDIR
+	echo ${MANDIR} > out/MANDIR
+	touch out/build-config
+
+out/genkernel.conf: out/build-config
+	cat genkernel.conf | sed \
+		-e '/#BEGIN FEATURES genkernel_conf/ r out/temp/genkernel_conf' \
+		> out/genkernel.conf
```

Comment:
This won't trigger properly on updates.
If I edit something in `genkernel.conf` and run make again, `out/genkernel.conf` will not be regenerated.

It's also unclear why it depends on `out/build-config` when it does NOT consume that file at all.

```
out/genkernel.conf: genkernel.conf out/temp/genkernel_conf
    cat $< | sed -e '/#BEGIN FEATURES genkernel_conf/ r out/temp/genkernel_conf' >$@
```

The same issue exists for many of your other targets.

I suggest creating a helper script that supports every transformation you've coded into the Makefile, and then simplifying the makefile:
```
out/genkernel.conf: genkernel.conf out/build-config $(INTERMEDIATE_DEPS) helper.sh
  ./helper.sh < $< > $@
```

and then you can see how it's easy to convert that rule further (this is a little harder without changing the layout to have the source files in a distinct directory).
```
out/%.conf: %.conf out/build-config $(INTERMEDIATE_DEPS) helper.sh
  ./helper.sh < $< > $@
```

Reply:
I created the build-config despite not being directly consumed because i was having problem getting the proper dependencies to register for preventing rebuilds. I thought about having separate dependencies for the temp files, but they are all processed at once within the awk script.